packages/main/src/AINotice.ts

```diff
@@ -0,0 +1,175 @@
+	 * @public
+	 */
+	@property()
+	verificationText = "";
+
+	/**
+	 * Defines whether an icon should be shown in the component.
+	 * @default false
+	 * @public
+	 */
+	@property({ type: Boolean })
+	showIcon = false;
+
+	/**
+	 * The popup slot allows for custom popups to be inserted into the component.
+	 * @public
+	 */
+	@slot({ type: HTMLElement })
+	popup!: HTMLElement;
+
+	@i18n("@ui5/webcomponents")
+	static i18nBundle: I18nBundle;
+
+	constructor() {
+		super();
+		this._handleResize = this.resizeWidth.bind(this);
+	}
+
+	onEnterDOM() {
+		ResizeHandler.register(this, this._handleResize);
+	}
+
+	onExitDOM() {
+		ResizeHandler.deregister(this, this._handleResize);
+	}
+
+	/**
+	 * A getter that determines whether the component is in the "emphasized" state, which includes attribution text, verification text, and an icon.
+	 */
+	get isEmphasized(): boolean {
+		return !!this.attributionText && !!this.verificationText && this.showIcon;
+	}
+
+	/**
+	 * A getter that determines whether the component is in the "icon-only" state, where only the icon is displayed.
+	 */
+	get isIconOnly(): boolean {
+		return !this.attributionText && !this.verificationText && this.showIcon;
+	}
+
+	/**
+	 * A getter that determines whether the component is in the "shortened" state, which shows only attribution text and hides verification text and icon.
+	 */
+	get isShortened(): boolean {
+		return !!this.attributionText && !this.verificationText && !this.showIcon;
+	}
+
+	get tooltipValue() {
+		return AINotice.i18nBundle.getText(AINOTICE_BUTTON_TOOLTIP);
+	}
+
+	/**
+	 * Resizes the component based on its content and the available space.
+	 * It adjusts the visibility of the verification text depending on the available width.
+	 */
+	resizeWidth() {
+		const container = this.getDomRef();
+		const verificationText = this.shadowRoot?.querySelector(".verificationText") as HTMLElement;
+		const attributionText = this.shadowRoot?.querySelector(".link-text") as HTMLElement;
+		if (container && verificationText && attributionText) {
+			const containerWidth = container.offsetWidth;
+			const verificationTextWidth = verificationText.offsetWidth;
+			const attributionTextWidth = attributionText ? attributionText.offsetWidth : 0;
+
+			const aInputRelevantCSS = ["padding-left"]; // Relevant CSS for padding
+			const computedStyle = window.getComputedStyle(verificationText);
+			const paddingLeft = parseInt(computedStyle.getPropertyValue(aInputRelevantCSS[0]) || "0"); // Fallback to "0" if no value is found
+			const combinedWidth = verificationTextWidth + attributionTextWidth + paddingLeft;
+
+			// Show or hide verification text based on available space
+			if (combinedWidth <= containerWidth) {
+				verificationText.style.visibility = "visible";
+			} else {
+				verificationText.style.visibility = "hidden";
+			}
+		}
+	}
+
+	/**
+	 * Handles the click event for the component and opens the associated popup.
+	 */
+	onPress(e: Event) {
```

Comment:
I don't get why we open this popover. It's pretty easy for developers to open it.
We only need to provide an event, and they can execute this on their side.

```ts
	popup.opener = e.target as HTMLElement;
	popup.open = true;
```

This way, the entire popup slot also will become obsolete.
Is there a specific reason to have a popup slot?